host/windows/syscall.c

```diff
@@ -3032,3 +3032,69 @@ int oe_syscall_nanosleep_ocall(struct oe_timespec* req, struct oe_timespec* rem)
     _set_errno(0);
     return 0;
 }
+
+/*
+**==============================================================================
+**
+** clock_gettime():
+**
+**==============================================================================
+*/
+
+/*
+**==============================================================================
+**
+** POSIX_TO_WINDOWS_EPOCH_TICKS:
+**
+** The ticks elapsed since 1970-01-01 00:00 (UTC). This value was derived with
+** the following function.
+**
+**     LONGLONG get_posix_to_windows_epoch_ticks()
+**     {
+**         SYSTEMTIME st = { .wYear = 1970, .wMonth = 1, .wDay = 1 };
+**         FILETIME ft;
+**         ULARGE_INTEGER x;
+**
+**         SystemTimeToFileTime(&st, &ft);
+**         x.u.LowPart = ft.dwLowDateTime;
+**         x.u.HighPart = ft.dwHighDateTime;
+**
+**         return x.QuadPart;
+**     }
+**
+**==============================================================================
+*/
+
+const LONGLONG POSIX_TO_WINDOWS_EPOCH_TICKS = 0X19DB1DED53E8000;
+
+/* Return milliseconds elapsed since the Epoch. */
+static void _time(struct oe_timespec* cur_time)
+{
+    FILETIME ft;
+    ULARGE_INTEGER x;
+    const LONGLONG TICKS_PER_MILLISECOND = 10000UL;
+    const LONGLONG MILLIS_PER_SEC = 1000UL;
+    const LONGLONG NANOS_PER_MILLI = 1000000UL;
+    GetSystemTimeAsFileTime(&ft);
+    x.u.LowPart = ft.dwLowDateTime;
+    x.u.HighPart = ft.dwHighDateTime;
+    x.QuadPart -= POSIX_TO_WINDOWS_EPOCH_TICKS;
+
+    uint64_t msec = (x.QuadPart / TICKS_PER_MILLISECOND);
+    cur_time->tv_sec = (msec / MILLIS_PER_SEC);
+    cur_time->tv_nsec = (msec % MILLIS_PER_SEC) * NANOS_PER_MILLI;
+}
+
+int oe_syscall_clock_gettime_ocall(int clock_id, struct oe_timespec* cur_time)
+{
+    if (!cur_time)
+        return -1;
+    memset(cur_time, 0, sizeof(struct oe_timespec));
+    if (clock_id != CLOCK_REALTIME)
```

Comment:
Given that we don't really need this check on Linux, can you check the impact on the Windows API we use (i.e., can we just remove them on Windows as well)?

Reply:
Looks like depending on the clock id we have to define separate functions on windows host. Current code only supports CLOCK_REALTIME. Hence above check should be present.

As per linux man pages, clock id take below four arguments. Should we support all this alternatives on windows host? Currently it implements only for CLOCK_REALTIME.

CLOCK_REALTIME
System-wide real-time clock. Setting this clock requires appropriate privileges. 

CLOCK_MONOTONIC
Clock that cannot be set and represents monotonic time since some unspecified starting point.

CLOCK_PROCESS_CPUTIME_ID
High-resolution per-process timer from the CPU.

CLOCK_THREAD_CPUTIME_ID
Thread-specific CPU-time clock.

Reply:
In this case, I'm fine with supporting only `CLOCK_REALTIME` for now and error out other options.  Can you add the comments to explain? Also, please update the systemedl document accordingly (i.e., the difference between Linux and Windows).